### DISCUSSION TITLE
Feat: Accent colour through GNOME Portal

### DISCUSSION THREAD

**Discussion by cyrneko** [YaLTeR/niri#891] [Ideas]
For a little bit now, we have had a proper accent color API that was implemented in KDE and eventually GNOME

Because we use the `xdg-desktop-portal-gnome` we should be able to pass accent colour information using it (as far as I understand) based on a value set in Niri's configuration, think maybe a `accent-color` that then accepts the same (non-gradient) values as the `focus-ring` or `border` do

Of course GNOME Apps don't accept just any color, from behavior I've seen on KDE it seems to choose the closest GNOME Color, so a megenta-ish color would become GNOME's purple, an ultramarine color would become GNOME's blue, etc

Maybe also a way to sync the accent with (one of) the `focus-ring` or `border` colors? Not sure how useful that'd be though...

Oh, and to get that out of the way, there is probably a way to do this with dbus that I simply don't know about, just thought proper config for this would be cool :woozy_face:

**Comment by YaLTeR** [comment]
Hey, there is indeed a dconf method:
```
$ dconf write /org/gnome/desktop/interface/accent-color "'yellow'"
```
(note the single quotes inside double quotes)

Or you could run gnome-control-center and change there:
```
$ XDG_CURRENT_DESKTOP=gnome gnome-control-center
```

I'm not sure I want to build this setting into niri because it is a GNOME setting which we need just because we happen to currently use the GNOME portal. Similarly to how there's no niri light-dark toggle, and instead you need to write to dconf by hand.

**Reply by cyrneko** [reply]
oh I didn't notice that dconf path existing :woozy_face: 

Though...I am pretty sure this does limit you to the GNOME colors, where things like e.g Plasma expose *any* color of the users' choice...but that just comes from my expectations of GNOME, I don't expect them to allow setting an arbitrary hex color value, and even with Plasma setting a fully custom one in GNOME apps it ends up being a different similar one

thanks for the dconf command though :eyes:

**Reply by YaLTeR** [reply]
You could probably set in niri-portals.conf to use the KDE portal for Settings. Then it'll use the KDE accent color setting.

**Comment by valpackett** [comment]
IMO it should be the other way around: use the dconf color in niri, for example with color functions:

```kdl
layout {
    focus-ring {
        active-gradient from="lighten(--gnome-accent-color, 0.1)" to="darken(--gnome-accent-color, 0.1)" angle=45
    }
}
```

Though in my crazy project of building an "actually GNOME session based" DE (like what Phosh does) something like that would be hardcoded anyway :upside_down_face:
